eZ/Publish/Core/FieldType/Image/ImageStorage/Gateway/LegacyStorage.php

```diff
@@ -236,6 +236,91 @@ public function countImageReferences($uri)
         return (int)$statement->fetchColumn();
     }
 
+    /**
+     * Returns the number of recorded references outside of the given $path.
+     *
+     * @param string $uri File IO uri (not legacy)
+     *
+     * @return int
+     */
+    public function countImageReferencesOutsidePath($uri)
+    {
+        $path = $this->redecorator->redecorateFromSource($uri);
+
+        $connection = $this->getConnection()->getConnection();
+
+        $queryBuilder = $connection->createQueryBuilder();
+        $statement = $queryBuilder
+            ->select('count(id)')
+            ->from('ezimagefile')
+            ->where('filepath not like :path')
+            ->setParameter('path', $path . '%')
+            ->execute();
+
+        return (int)$statement->fetchColumn();
+    }
+
+    /**
+     * Updates the filepath of given Image.
+     *
+     * @param $imageId
+     * @param $newFilePath
```

Comment:
```suggestion
     * @param string $newFilePath
```

Reply:
Scalar type declarations were introduced in php 7.0 and kernel 6.13 supports php 5.6